packages/datadog-instrumentations/src/mysql2.js

```diff
@@ -79,3 +152,138 @@ addHook({ name: 'mysql2', file: 'lib/connection.js', versions: ['>=1'] }, Connec
     }, cmd))
   }
 })
+
+addHook({ name: 'mysql2', file: 'lib/pool.js', versions: ['>=1'] }, (Pool, version) => {
+  const startOuterQueryCh = channel('datadog:mysql2:outerquery:start')
+  const shouldEmitEndAfterQueryAbort = semver.intersects(version, '>=1.3.3')
+
+  shimmer.wrap(Pool.prototype, 'query', query => function (sql, values, cb) {
+    if (!startOuterQueryCh.hasSubscribers) return query.apply(this, arguments)
+
+    const sqlString = typeof sql === 'object' ? sql?.sql : sql
+    if (sqlString) {
+      const abortController = new AbortController()
+      startOuterQueryCh.publish({ sql, abortController })
+
+      if (abortController.signal.aborted) {
+        const getConnection = this.getConnection
+        this.getConnection = function () {}
+
+        let queryCommand
+        try {
+          queryCommand = query.apply(this, arguments)
+        } finally {
+          this.getConnection = getConnection
+        }
+
+        process.nextTick(() => {
+          if (queryCommand.onResult) {
+            queryCommand.onResult(abortController.signal.reason)
+          } else {
+            queryCommand.emit('error', abortController.signal.reason)
+          }
+
+          if (shouldEmitEndAfterQueryAbort) {
+            queryCommand.emit('end')
+          }
+        })
+
+        return queryCommand
+      }
+
+      return query.apply(this, arguments)
+    }
+  })
+
+  shimmer.wrap(Pool.prototype, 'execute', execute => function (sql, values, cb) {
+    if (!startOuterQueryCh.hasSubscribers) return execute.apply(this, arguments)
+
+    const abortController = new AbortController()
+    startOuterQueryCh.publish({ sql, abortController })
```

Comment:
In some other places, a check is done to assure there is actually a query (like [here](https://github.com/DataDog/dd-trace-js/pull/4712/files#diff-30989a4234a7039a68994bf4e35c149c3ef115c24b65682c793ce6301467168bR164)). Is there any reason to not do this check here? (and [here](https://github.com/DataDog/dd-trace-js/pull/4712/files#diff-30989a4234a7039a68994bf4e35c149c3ef115c24b65682c793ce6301467168bR164) and [here](https://github.com/DataDog/dd-trace-js/pull/4712/files#diff-30989a4234a7039a68994bf4e35c149c3ef115c24b65682c793ce6301467168bR164))

Reply:
I changed all the cases to not call to publish without sql, just to prevent creating unnecessary AbortControllers.